src/api-service/__app__/onefuzzlib/proxy.py

```diff
@@ -158,6 +179,9 @@ def extensions_launch(self) -> None:
 
     def stopping(self) -> None:
         vm = self.get_vm()
```

Comment:
This feels weird, considering `get_vm` comment earlier, where during the VM deletion process, we'll create an NSG, then delete it... is that intended?

Reply:
NSG does not get deleted when VM is deleted, VM's NIC is dissociated from NSG. To avoid a window where VM is in the open - we need to dissociate right before NIC gets deleted.

Reply:
No, NIC has to be dissociated first. 
But even if NIC is available - the actual virtual machine is deleted prior to the point where we reach NIC deletion (line 195). 
So worse case we end up with stand alone "public" NIC without a VM